### PR TITLE
add recoil pitch from Crispy Doom

### LINKED ISSUE #503: add weapon recoil pitch
Probably a direct port of the Crispy feature.

### PR DESCRIPTION
**Pull request by rfomin**
Fixes #503

### CODE REVIEW

File: Source/m_menu.c [hunk 1]
```diff
@@ -3016,7 +3016,6 @@ static const char *weapon_attack_alignment_strings[] = {
 
 setup_menu_t weap_settings1[] =  // Weapons Settings screen       
 {
-  {"ENABLE RECOIL", S_YESNO,m_null,WP_X, WP_Y+ weap_recoil*8, {"weapon_recoil"}},
```
Comment (fabiangreffrath) [inline Source/m_menu.c:3019]: So, we are really kicking this feature out of the menu?

File: Source/p_pspr.c [hunk 2]
```diff
@@ -737,7 +753,7 @@ void A_FireOldBFG(player_t *player, pspdef_t *psp)
 
   if (weapon_recoil && !(player->mo->flags & MF_NOCLIP))
     P_Thrust(player, ANG180 + player->mo->angle,
-	     512*recoil_values[wp_plasma]);
+	     512*recoil_values[wp_plasma].thrust);
```
Comment (fabiangreffrath) [inline Source/p_pspr.c:756]: Any chance to implement recoil pitch for the "old" BFG similar to the plasma gun?

File: Source/m_menu.c (hunk 1 above)
Comment (rfomin) [reply Source/m_menu.c:3019]: This demo is incompatible and I doubt anyone will complain. Crispy recoil is much better.

File: Source/p_pspr.c (hunk 2 above)
Comment (rfomin) [reply Source/p_pspr.c:758]: Hm, I tried in https://github.com/fabiangreffrath/woof/pull/506/commits/d095cbc33ce22de8928e0b66741a076cce259773 , but it's not satisfying.

File: Source/r_main.c [hunk 3]
```diff
@@ -572,25 +572,34 @@ void R_SetupFrame (player_t *player)
...
   viewy = player->mo->y;
   viewz = player->viewz; // [FG] moved here
   viewangle = player->mo->angle + viewangleoffset;
-  lookdir = player->lookdir / MLOOKUNIT;
+  // [crispy] pitch is actual lookdir and weapon pitch
+  pitch = player->lookdir / MLOOKUNIT + player->recoilpitch;
   }
   extralight = player->extralight;
     
+
```
Comment (fabiangreffrath) [inline Source/r_main.c:590]: excess blank line

File: Source/m_menu.c (hunk 1 above)
Comment (fabiangreffrath) [reply Source/m_menu.c:3019]: But demos recorded with this feature enabled (would) still play back correctly, right?

Comment (rfomin) [reply Source/m_menu.c:3019]: Right. I guess we can make both available, but one affects gameplay and the other is purely cosmetic.

Comment (fabiangreffrath) [reply Source/m_menu.c:3019]: I think I'd prefer that. It's one of the original MBF fatures after all.

Didn't we want to rearrange that part of the menu anyway? 😉

File: Source/m_menu.c [hunk 4]
```diff
@@ -2996,16 +2996,20 @@ enum {           // killough 10/98: enum for y-offset info
   weap_stub2,
   weap_toggle,
   weap_toggle2,
-  weap_stub3,
+};
+
+enum {
   weap_consmetic,
```
Comment (fabiangreffrath) [inline Source/m_menu.c:3002]: typo

File: Source/m_menu.c [hunk 5]
```diff
@@ -6471,7 +6488,6 @@ void M_ResetSetupMenu(void)
     enem_settings1[13].m_flags |= S_DISABLE;
 
   FLAG_SET_VANILLA(enem_settings1[enem_remember].m_flags, S_DISABLE);
-  FLAG_SET_VANILLA(weap_settings1[weap_recoil].m_flags, S_DISABLE);
```
Comment (fabiangreffrath) [inline Source/m_menu.c:6474]: doesn't this have to be restored now?

Comment (rfomin) [reply Source/m_menu.c]: Fixed in https://github.com/fabiangreffrath/woof/pull/506/commits/afa3b1b44796848aae8ed53509aea8e20a0b2248

File: Source/p_pspr.c (hunk 2 above)
Comment (fabiangreffrath) [reply Source/p_pspr.c:758]: > Hm, I tried in [d095cbc](https://github.com/fabiangreffrath/woof/commit/d095cbc33ce22de8928e0b66741a076cce259773) , but it's not satisfying.

How about this?

```patch
--- a/Source/p_pspr.c
+++ b/Source/p_pspr.c
@@ -757,7 +757,7 @@ void A_FireOldBFG(player_t *player, pspdef_t *psp)
     P_Thrust(player, ANG180 + player->mo->angle,
             512*recoil_values[wp_plasma].thrust);
 
-  if (weapon_recoilpitch)
+  if (weapon_recoilpitch && (leveltime & 2))
   {
     player->recoilpitch = recoil_values[wp_plasma].pitch;
   }
```

Comment (rfomin) [reply Source/p_pspr.c:758]: This is better! Added in https://github.com/fabiangreffrath/woof/pull/506/commits/ff713ebf51453e6dc4d0d528d8fb144c8b5598c6

**Review by fabiangreffrath — APPROVED** [review]

**Pull request merged**